0003-siren-prototype-accept-decline.adoc

```diff
@@ -0,0 +1,88 @@
+= Siren Prototype for comit-rs HTTP API - Accept/Decline actions
+
+* Authors: Franck Royer
+* Date: 2019-05-02
+
+Tracking issue: https://github.com/comit-network/comit-rs/issues/917[comit-rs#917]
+
+== Context
+
+In order to move forward easily with Bobtimus prime we should define the siren json document of how a IN_PROGRESS response look like in each state of accept and decline, i.e. when you call GET /swaps/ for Bob.
+
+Siren repository: https://github.com/kevinswiber/siren/
+
+The prototype must conform to https://github.com/kevinswiber/siren/blob/master/siren.schema.json[Siren JSON schema]
+
+== Research
+
+.The prototype
+[source,json]
+----
+{
+  "class": ["swaps"],
+  "entities": [
+    {
+      "class": ["swap"],
+      "rel": ["item"],
+      "properties": {
+        "role": "Alice",
+        "protocol": "rfc003",
+        "status": "IN_PROGRESS",
+        "parameters": {
+          "alpha_asset": {
+            "name": "bitcoin",
+            "quantity": "100000000"
+          },
+          "alpha_ledger": {
+            "name": "bitcoin",
+            "network": "regtest"
+          },
+          "beta_asset": {
+            "name": "ether",
+            "quantity": "10000000000000000000"
+          },
+          "beta_ledger": {
+            "name": "ethereum",
+            "network": "regtest"
+          }
+        }
+      },
+      "links": [
+        {"rel": ["self"], "href": "/swaps/rfc003/399e8ff5-9729-479e-aad8-49b03f8fc5d5"},
+        {
+          "rel": ["human-protocol-spec"],
+          "href": "https://github.com/comit-network/RFCs/blob/master/RFC-003-SWAP-Basic.md"
+        }
+      ],
+      "actions": [
+        {
+          "name": "accept",
+          "title": "Accept",
+          "method": "POST",
+          "href": "/swaps/rfc003/399e8ff5-9729-479e-aad8-49b03f8fc5d5/accept",
+          "type": "application/json",
+          "class": ["comit-action"],
```

Comment:
I agree with @bonomat this is useless.

Reply:
This is actually not useless at all!
You need a way of identifying, which actions should be rendered as buttons to the user. Assuming that all actions of a swap go in the _same_ list of choices is a bit dangerous IMO.

Reply:
I see no added value to `"comit-action"`. The fact that it is a `POST` is enough to know that a button (to trigger the `POST`) should be rendered to the user.

Reply:
My concern is about forward compatibility. If we don't provide the class `comit-action`, clients will default to rendering all the actions of a swap next to each other. Currently, this is what we want. My thinking was that there may be SIREN actions at some point that are of a different kind other than the ones we currently have and thus we may want handle them differently from the current ones (render them in a different place for example).

Without a `comit-action` class, those will have to be explicitly filtered out whereas, with a `comit-action` class, the code that renders the current actions will always work, regardless of what we do in the future.

You could argue that we are never going to add any actions that need to be treated differently. Maybe that is the case, maybe not.

Thinking about it, it is probably too much premature optimization :)

Reply:
I see your point and I don't think I have a strong opinion anymore :)

Reply:
@thomaseizinger : I'm not 100% sure I understand your thoughts:
if the client receives a POST action with some `fields`, then it should know that a form is required to collect the data (if it is an interactive client). 
I assume this should always be like that for all actions with some `fields`.
If there are no fields, then it's a simple button (e.g. the decline action). 

The way how I would interpret this is: if it is a `comit-action` render a `comit-logo` on top of the form/button/etc, if `ethereum-action`, render an `ethereum-logo`...

--> Happy to put it there for now .

Reply:
> @thomaseizinger : I'm not 100% sure I understand your thoughts:
> if the client receives a POST action with some `fields`, then it should know that a form is required to collect the data (if it is an interactive client).
> I assume this should always be like that for all actions with some `fields`.
> If there are no fields, then it's a simple button (e.g. the decline action).
> 
> The way how I would interpret this is: if it is a `comit-action` render a `comit-logo` on top of the form/button/etc, if `ethereum-action`, render an `ethereum-logo`...
> 
> --> Happy to put it there for now .

Best to hash it out face-to-face?

Reply:
Discussed it offline - happy to remove it.